x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -67,6 +63,7 @@ public class JwtRealm extends Realm implements CachingRealm, Releasable {
     public static final String HEADER_CLIENT_AUTHENTICATION = "ES-Client-Authentication";
     public static final String HEADER_END_USER_AUTHENTICATION_SCHEME = "Bearer";
     public static final String HEADER_SHARED_SECRET_AUTHENTICATION_SCHEME = "SharedSecret";
+    private static final ThreadLocal<SecureString> latestMalformedJWTThreadLocal = new ThreadLocal<>();
```

Comment:
Can we use ThreadContext#putTransient instead of creating our own ThreadLocal ?  The ThreadContext provides guarantees of request scoping as well as guarantees to persist across forked threads in case of concurrency (assuming you use our custom executors).  Since this work is happening on the transport threads and we use thread pools I am pretty sure this ThreadLocal can bleed between requests.

Reply:
Pushed [3316600](https://github.com/elastic/elasticsearch/pull/101333/commits/331660081de79d5b6ae51f8c08b85c073d1e53c7) .